recipes/nettle/all/conanfile.py

```diff
@@ -47,84 +49,114 @@ def config_options(self):
-        tools.replace_in_file(makefile_in,
+        makefile_in = os.path.join(self.source_folder, "Makefile.in")
+        # discard subdirs
+        replace_in_file(self, makefile_in,
                               "SUBDIRS = tools testsuite examples",
                               "SUBDIRS = ")
         # Fix broken tests for compilers like apple-clang with -Werror,-Wimplicit-function-declaration
-        tools.replace_in_file(os.path.join(self._source_subfolder, "aclocal.m4"),
+        replace_in_file(self, os.path.join(self.source_folder, "aclocal.m4"),
                               "cat >conftest.c <<EOF",
                               "cat >conftest.c <<EOF\n#include <stdlib.h>")
 
     def build(self):
         self._patch_sources()
-        with tools.chdir(self._source_subfolder):
-            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows)
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
+        autotools.autoreconf()
+        # srcdir in unix path causes some troubles in asm files on Windows
+        if self.settings.os == "Windows":
+            replace_in_file(self, os.path.join(self.build_folder, "config.m4"),
+                                  unix_path(self, os.path.join(self.build_folder, self.source_folder)),
+                                  os.path.join(self.build_folder, self.source_folder).replace("\\", "/"))
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy(pattern="COPYING*", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        copy(self, pattern="COPYING*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        autotools = Autotools(self)
         autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.components["hogweed"].names["pkgconfig"] = "hogweed"
+        self.cpp_info.set_property("cmake_file_name", "Nettle")
+
+        self.cpp_info.components["hogweed"].set_property("pkg_config_name", "hogweed")
+        self.cpp_info.components["hogweed"].set_property("cmake_target_name", "Nettle::Hogweed")
         self.cpp_info.components["hogweed"].libs = ["hogweed"]
         if self.options.public_key:
             self.cpp_info.components["hogweed"].requires.append("gmp::libgmp")
 
-        self.cpp_info.components["libnettle"].libs = ["nettle"]
-        self.cpp_info.components["libnettle"].requires = ["hogweed"]
-        self.cpp_info.components["libnettle"].names["pkgconfig"] = "nettle"
+        self.cpp_info.components["nettle"].libs = ["nettle"]
```

Comment:
why it it possible now to use recipe name as component name?

Reply:
Didn't since when it's possible, but indeed is not a good idea.